editor/app/ems/[incident_number]/page.tsx

```diff
@@ -0,0 +1,205 @@
+"use client";
+import { useCallback, useEffect, useMemo, useState } from "react";
+import { notFound } from "next/navigation";
+import Col from "react-bootstrap/Col";
+import Row from "react-bootstrap/Row";
+import DataCard from "@/components/DataCard";
+import { emsDataCards } from "@/configs/emsDataCards";
+import { useMutation, useQuery } from "@/utils/graphql";
+import {
+  GET_EMS_RECORDS,
+  GET_MATCHING_PEOPLE,
+  UPDATE_EMS_INCIDENT,
+  UPDATE_EMS_INCIDENT_CRASH_AND_PERSON,
+} from "@/queries/ems";
+import { EMSPatientCareRecord } from "@/types/ems";
+import RelatedRecordTable from "@/components/RelatedRecordTable";
+import EMSLinkRecordButton, {
+  EMSLinkRecordButtonProps,
+} from "@/components/EMSLinkRecordButton";
+import EMSLinkToPersonButton, {
+  EMSLinkToPersonButtonProps,
+} from "@/components/EMSLinkToPersonButton";
+import { emsMatchingPeopleColumns } from "@/configs/emsMatchingPeopleColumns";
+import { PeopleListRow } from "@/types/peopleList";
+import { FaTruckMedical } from "react-icons/fa6";
+
+export default function EMSDetailsPage({
+  params,
+}: {
+  params: { incident_number: string };
+}) {
+  const [selectedEmsPcr, setSelectedEmsPcr] =
+    useState<EMSPatientCareRecord | null>(null);
+
+  const incident_number = params.incident_number;
+
+  /**
+   * Get all EMS records associated with this incident
+   */
+  const {
+    data: ems_pcrs,
+    error,
+    isValidating,
+    refetch,
+  } = useQuery<EMSPatientCareRecord>({
+    query: incident_number ? GET_EMS_RECORDS : null,
+    variables: {
+      incident_number: incident_number,
+    },
+    typename: "ems__incidents",
+  });
+
+  const { mutate: updateEMSIncident } = useMutation(
+    UPDATE_EMS_INCIDENT_CRASH_AND_PERSON
+  );
+
+  /**
+   * Hook which manages which related crash PKs we should
+   * use to query people records
+   */
+  const relatedCrashPks: number[] | null = useMemo(() => {
+    if (!ems_pcrs) {
+      return null;
+    }
+    const relatedCrashPks: number[] = [];
```

Comment:
at first it confused me that this var was being created twice and i was gonna suggest to just use a .map() but it makes sense how you did it with having to dedupe and all that :+1: